pre-docs/writing_tasks.md

```diff
@@ -274,6 +274,120 @@ myservice/tasks/stop.json
 }
 ```
 
+## Sharing task code
+
+Multiple tasks can share common files between them. Tasks can additionally pull library code from other modules.
+
+To create a task that includes additional files pulled from modules, include the `files` property in your metadata as an array of paths. A path consists of
+- the module name
+- one of `lib`, `files`, or `tasks` for the directory within the module
+- the remaining path to a file or directory; directories must include a trailing slash `/`
+
+All path separators must be forward slashes. An example would be `stdlib/lib/puppet/`.
+
+The `files` property can be included both as a top-level metadata property, and as a property of an implementation, for example
+```
+{
+  "implementations": [
+    {"name": "sql.sh", "requirements": ["shell"], "files": ["mymodule/files/lib.sh"]},
+    {"name": "sql.ps1", "requirements": ["powershell"], "files": ["mymodule/files/lib.ps1"]}
+  ],
+  "files": ["emoji/files/emojis/"]
+}
+```
+
+When a task includes the `files` property, all files listed in the top-level property and in the specific implementation chosen for a target will be copied to a temporary directory on that target. The directory structure of the specified files will be preserved such that paths specified with the `files` metadata option will be available to tasks prefixed with `_installdir`.
+
+### Python Example
+
+#### Metadata
+
+```json
+{
+  "files": ["multi_task/files/py_util/py_helper.py"]
+}
+```
+
+#### Files
+
+`multi_task/files/py_util/py_helper.py`
+
+```python
+def useful_python():
+  return dict(helper="python")
+```
+
+#### Task
+
+```python
+#!/usr/bin/env python
+import sys
+import os
+import json
+
+params = json.load(sys.stdin)
+sys.path.append(os.path.join(params['_installdir'], 'python_helpers/files'))
```

Comment:
does this need to be 
os.path.join(params['_installdir'], 'python_helpers', 'files')